arch/src/lib.rs

```diff
@@ -59,6 +59,14 @@ pub enum Error {
 /// Type for returning public functions outcome.
 pub type Result<T> = result::Result<T, Error>;
 
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+pub enum CpuProfile {
```

Comment:
I much rather have:

`arch/{mod, x86_64, aarch64, risv64}.rs` each with their own `pub enum CpuProfile`.

On the top level, you could do still something like:

```rust
pub enum CpuProfile {
  Host,
  Specific(arch::CpuProfile)
}
```